build-system/tasks/prettify.js

```diff
@@ -0,0 +1,144 @@
+const {logOnSameLine} = require('../common/utils');
+const {prettifyGlobs} = require('../test-configs/config');
+
+const rootDir = path.dirname(path.dirname(__dirname));
+const prettierCmd = 'node_modules/.bin/prettier';
+
+/**
+ * Prettier exit codes. See https://prettier.io/docs/en/cli.html#exit-codes
+ * @enum {number}
+ */
+const PrettierResult = {
+  SUCCESS: 0,
+  FAILURE: 1,
+  ERROR: 2,
+};
+
+/**
+ * Checks files for formatting (and optionally fixes them) with Prettier.
+ * Returns the cumulative result to the `gulp` process via  process.exitCode so
+ * that all files can be checked / fixed.
+ *
+ * @return {!Promise}
+ */
+function prettify() {
+  const filesToCheck = argv.files ? argv.files.split(',') : prettifyGlobs;
+  return gulp
+    .src(filesToCheck)
+    .pipe(
+      tap(file => {
+        checkFile(path.relative(rootDir, file.path));
+      })
+    )
+    .on('finish', () => {
+      if (process.exitCode == 1) {
+        logOnSameLine(red('ERROR: ') + 'Found errors in one or more files.');
+        if (!argv.fix) {
+          log(
+            yellow('NOTE 1:'),
+            'You may be able to automatically fix some of these warnings ' +
+              '/ errors by running',
+            cyan('gulp prettify --fix'),
+            'from your local branch.'
+          );
+          log(
+            yellow('NOTE 2:'),
+            'Since this is a destructive operation (that edits your files',
+            'in-place), make sure you commit before running the command.'
+          );
+        }
+      } else {
+        if (!isTravisBuild()) {
+          logOnSameLine(green('SUCCESS: ') + 'No formatting errors found.');
+        }
+      }
+    });
+}
+
+/**
+ * Fixes the formatting of a single file
+ * @param {string} file
+ */
+function fixFile(file) {
+  const fixCmd = `${prettierCmd} --write ${file}`;
+  const fixResult = getOutput(fixCmd);
+  if (fixResult.status == PrettierResult.SUCCESS) {
+    logOnSameLine(green('Fixed: ') + file + '\n');
+  } else {
+    logOnSameLine(red('Could not fix: ') + file);
+    console.log(highlight(fixResult.stderr, {ignoreIllegals: true}), '\n');
+    process.exitCode = 1;
+  }
+}
+
+/**
+ * Checks and optionally fixes the formatting of a single file.
+ * @param {string} file
+ */
+function checkFile(file) {
+  const checkCmd = `${prettierCmd} --list-different ${file}`;
+  const checkResult = getOutput(checkCmd);
+  if (checkResult.status == PrettierResult.SUCCESS) {
+    if (!isTravisBuild()) {
+      logOnSameLine(green('Checked: ') + file);
+    }
+  } else if (checkResult.status == PrettierResult.FAILURE) {
+    if (argv.fix) {
+      fixFile(file);
+    } else {
+      logOnSameLine(red('Found errors in: ') + file + '\n');
+      process.exitCode = 1;
+    }
+  } else {
+    logOnSameLine(red('Could not parse: ') + file);
+    console.log(
+      highlight(checkResult.stderr.trim(), {ignoreIllegals: true}),
+      '\n'
+    );
+    process.exitCode = 1;
```

Comment:
Maybe just `process.exitCode = checkResult;` right after `getOutput` here and above to just propagate the Prettier result

Reply:
Tried that first, it got complicated because there is a legit case where one `getOutput` (check) can fail and another (fix) can pass, and the result is a success.

I'll try to clean this up when I go about adding `--local_changes` functionality.